storage/src/main/java/tech/pegasys/teku/storage/protoarray/ProtoArray.java

```diff
@@ -695,9 +682,11 @@ private boolean hasAncestorAtSlot(
     return node != null && requiredRoot.equals(node.getBlockRoot());
   }
 
-  private boolean doesCheckpointEpochMatch(final Checkpoint actual, final Checkpoint required) {
-    return required.getEpoch().equals(initialEpoch)
-        || actual.getEpoch().equals(required.getEpoch());
+  private boolean doesCheckpointEpochMatchOrIsRecentEnough(
+      final UInt64 actualEpoch, final UInt64 requiredEpoch) {
+    return requiredEpoch.equals(initialEpoch)
```

Comment:
i'd probably at least suggest comparison to initialEpoch is last, it's valid once...